api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -65,4 +76,143 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
 		}
 		
 	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)} 
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldHaveMultipleOrderGroups() {
+		OrderGroup firstOrderGroup = new OrderGroup();
+		OrderGroup secondOrderGroup = new OrderGroup();
+		OrderGroup thirdOrderGroup = new OrderGroup();
+
+		Encounter newEncounter1 = encounterService.getEncounter(3);
+		Encounter newEncounter2 = encounterService.getEncounter(4);
+
+		firstOrderGroup.setEncounter(newEncounter1);
+		secondOrderGroup.setEncounter(newEncounter1);
+		thirdOrderGroup.setEncounter(newEncounter2);
+
+		assertEquals(0, dao.getOrderGroupsByEncounter(newEncounter1).size());
+		assertEquals(0, dao.getOrderGroupsByEncounter(newEncounter2).size());
+		
+		dao.saveOrderGroup(firstOrderGroup);
+		dao.saveOrderGroup(secondOrderGroup);
+		dao.saveOrderGroup(thirdOrderGroup);
+
+		assertEquals(2, dao.getOrderGroupsByEncounter(newEncounter1).size());
+		assertEquals(1, dao.getOrderGroupsByEncounter(newEncounter2).size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldHaveNoOrderGroups() {
+		OrderGroup orderGroup = new OrderGroup();
+		
+		Encounter newEncounter1 = encounterService.getEncounter(3);
```

Comment:
Since this Encounter and in many other places is already existing in the DB, why don't we name it to something like `existingEncounter1`?

Reply:
Yeahh sounds like a good idea. Lmk if there are any other issues in my code.